cruise-control-core/src/main/java/com/linkedin/cruisecontrol/monitor/sampling/MetricSample.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2017 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
+ */
+
+package com.linkedin.cruisecontrol.monitor.sampling;
+
+import com.linkedin.cruisecontrol.metricdef.MetricDef;
+import com.linkedin.cruisecontrol.metricdef.MetricInfo;
+import com.linkedin.cruisecontrol.model.Entity;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class MetricSample<G, E extends Entity<G>> {
+  private static final Logger LOG = LoggerFactory.getLogger(MetricSample.class);
+  protected final E _entity;
+  protected final Map<Integer, Double> _metrics;
```

Comment:
If all the metrics are defined already then max id should be known in which case this can just be Double[] instead of map.  If you are willing to use NaN or some other placeholder to represent null values then this can be double[].

Reply:
These are short-lived objects so we do not care about memory consumption by that much. Also because the metric definitions we have is not guaranteed to be contiguous, a map is more flexible.